replica/table.cc

```diff
@@ -147,6 +147,25 @@ table::find_row(schema_ptr s, reader_permit permit, const dht::decorated_key& pa
     });
 }
 
+void
+table::add_memtables_to_reader_list(std::vector<flat_mutation_reader_v2>& readers,
+        const schema_ptr& s,
+        const reader_permit& permit,
+        const dht::partition_range& range,
+        const query::partition_slice& slice,
+        const io_priority_class& pc,
+        const tracing::trace_state_ptr& trace_state,
+        streamed_mutation::forwarding fwd,
+        mutation_reader::forwarding fwd_mr) const {
+    readers.reserve(readers.size() + _compaction_group->memtables()->size() + 1);
```

Comment:
I don't think this reservation will help anything in the normal case as typically compaction groups will have zero to one memtables

Reply:
reserving too little might cause high write amplification of the record every time the re-reservation adds just enough extra space. See 7f80602b010a973dd24953cf39161699e7afcb42